test/afalgtest.c

```diff
@@ -56,6 +58,19 @@ static int test_afalg_aes_128_cbc(void)
     int encl, encf, decl, decf;
     int ret = 0;
 
+    switch (keysize_idx) {
+        case 0:
+            cipher = EVP_aes_128_cbc();
+            break;
+        case 1:
+            cipher = EVP_aes_192_cbc();
+            break;
+        case 2:
+            cipher = EVP_aes_256_cbc();
+            break;
+        default:
+            cipher = NULL;
+    }
     if (!TEST_ptr(ctx = EVP_CIPHER_CTX_new()))
             return 0;
     RAND_bytes(in, BUFFER_SIZE);
```

Comment:
The problem with this test at the moment is that it doesn't demonstrate that the encryption actually worked. All it does is show that it is reversible, i.e. if you decrypt what you encrypted you get back to where you started. It would be good to have some evidence to show that we really are getting 128-bit, 192-bit or 256-bit AES-CBC here. For example, if something about your changes was not correct and we were getting 128-bit encryption in all cases in error - we would not know as a result of this test. Perhaps having some "known good" values (not originally generated via the afalg engine itself) for a given fixed (i.e. non random as now) input might be the answer?